cli/azd/pkg/project/ai_helper.go

```diff
@@ -0,0 +1,713 @@
+	if a.initialized {
+		return nil
+	}
+
+	if err := a.pythonBridge.Initialize(ctx); err != nil {
+		return err
+	}
+
+	a.initialized = true
+	return nil
+}
+
+// EnsureWorkspace ensures that the workspace exists
+func (a *aiHelper) EnsureWorkspace(
+	ctx context.Context,
+	scope *ai.Scope,
+) error {
+	workspaceName := scope.Workspace()
+
+	workspaceResponse, err := a.workspacesClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		workspaceName,
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	if *workspaceResponse.Workspace.Name != workspaceName {
+		return err
+	}
+
+	return nil
+}
+
+// CreateEnvironmentVersion creates a new environment version
+func (a *aiHelper) CreateEnvironmentVersion(
+	ctx context.Context,
+	scope *ai.Scope,
+	serviceConfig *ServiceConfig,
+	config *ai.ComponentConfig,
+) (*armmachinelearning.EnvironmentVersion, error) {
+	yamlFilePath := filepath.Join(serviceConfig.Path(), config.Path)
+	_, err := os.Stat(yamlFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	environmentName, err := config.Name.Envsubst(a.env.Getenv)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing environment name value: %w", err)
+	}
+
+	if environmentName == "" {
+		environmentName = fmt.Sprintf("%s-environment", serviceConfig.Name)
+	}
+
+	nextVersion := "1"
+	envContainerResponse, err := a.envContainersClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		scope.Workspace(),
+		environmentName,
+		nil,
+	)
+	if err == nil {
+		nextVersion = *envContainerResponse.Properties.NextVersion
+	}
+
+	environmentArgs := []string{
+		"-t", "environment",
+		"-s", scope.SubscriptionId(),
+		"-g", scope.ResourceGroup(),
+		"-w", scope.Workspace(),
+		"-f", yamlFilePath,
+		"--set", fmt.Sprintf("name=%s", environmentName),
+		"--set", fmt.Sprintf("version=%s", nextVersion),
+	}
+
+	environmentArgs, err = a.applyOverrides(environmentArgs, config.Overrides)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := a.pythonBridge.Run(ctx, ai.MLClient, environmentArgs...); err != nil {
+		return nil, err
+	}
+
+	envVersionResponse, err := a.envVersionsClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		scope.Workspace(),
+		environmentName,
+		nextVersion,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	a.env.DotenvSet(AiEnvironmentEnvVarName, environmentName)
+
+	return &envVersionResponse.EnvironmentVersion, nil
+}
+
+// CreateModelVersion creates a new model version
+func (a *aiHelper) CreateModelVersion(
+	ctx context.Context,
+	scope *ai.Scope,
+	serviceConfig *ServiceConfig,
+	config *ai.ComponentConfig,
+) (*armmachinelearning.ModelVersion, error) {
+	yamlFilePath := filepath.Join(serviceConfig.Path(), config.Path)
+	_, err := os.Stat(yamlFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	modelName, err := config.Name.Envsubst(a.env.Getenv)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing model name value: %w", err)
+	}
+
+	if modelName == "" {
+		modelName = fmt.Sprintf("%s-model", serviceConfig.Name)
```

Comment:
same thing about checking naming constrains here

Reply:
I'll look into it.

Reply:
Docs don't define any naming rules for model or environment names but do for deployment names. 

**Model Naming**
https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-model?view=azureml-api-2

**Environment Naming**
https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-environment?view=azureml-api-2

**Deployment Naming**
https://learn.microsoft.com/en-us/azure/machine-learning/how-to-manage-quotas?view=azureml-api-2#azure-machine-learning-online-endpoints-and-batch-endpoints